core/src/main/java/org/stellar/anchor/sep10/Sep10Service.java

```diff
@@ -60,9 +60,10 @@ public ChallengeResponse createChallenge(ChallengeRequest challengeRequest) thro
     // Validations
     //
     if (challengeRequest.getHomeDomain() == null) {
-      debugF("home_domain is not specified. Will use the default: {}", sep10Config.getHomeDomain());
-      challengeRequest.setHomeDomain(sep10Config.getHomeDomain());
-    } else if (!sep10Config.getHomeDomain().equalsIgnoreCase(challengeRequest.getHomeDomain())) {
+      debugF(
+          "home_domain is not specified. Will use the default: {}", sep10Config.getWebAuthDomain());
+      challengeRequest.setHomeDomain(sep10Config.getWebAuthDomain());
+    } else if (!sep10Config.getWebAuthDomain().equalsIgnoreCase(challengeRequest.getHomeDomain())) {
```

Comment:
`home_domain` needs to match the domain of `getHostUrl()`, which is where the toml is hosted.

Reply:
Fixed.